vulture_os/toolkit/api_parser/sentinel_one_mobile/sentinel_one_mobile.py

```diff
@@ -0,0 +1,201 @@
+from django.conf import settings
+from django.utils import timezone
+from toolkit.api_parser.api_parser import ApiParser
+
+logging.config.dictConfig(settings.LOG_SETTINGS)
+logger = logging.getLogger('api_parser')
+
+
+class SentinelOneMobileAPIError(Exception):
+    pass
+
+class SentinelOneMobileParser(ApiParser):
+    ALERT_URI = "api/v1/devices/public/device_updates"
+
+    def __init__(self, data):
+        super().__init__(data)
+
+        self.sentinel_one_mobile_host = data["sentinel_one_mobile_host"]
+        if not self.sentinel_one_mobile_host.startswith('https://'):
+            self.sentinel_one_mobile_host = f"https://{self.sentinel_one_mobile_host}"
+
+        self.sentinel_one_mobile_apikey = data["sentinel_one_mobile_apikey"]
+
+        self.session = None
+
+    def _connect(self):
+        try:
+            if self.session is None:
+                self.session = requests.Session()
+        except Exception as err:
+            raise SentinelOneMobileAPIError(err)
+        else:
+            return True
+
+    def execute_query(self, method, url, query=None, sleep_retry=10, timeout=10):
+        self._connect()
+
+        headers = {
+            "api_key": self.sentinel_one_mobile_apikey,
+            'Accept': 'application/json'
+        }
+
+        response = None
+        retry = 3
+
+        while retry > 0:
+            retry -= 1
+
+            try:
+                response = self.session.request(
+                    method=method,
+                    url=url,
+                    params=query,
+                    headers=headers,
+                    proxies=self.proxies,
+                    timeout=timeout
+                )
+            except requests.exceptions.ReadTimeout:
+                msg = f"ReadTimeout, waiting {sleep_retry}s before retrying"
+                logger.info(f"[{__parser__}]:execute_query: {msg}", extra={'frontend': str(self.frontend)})
+                time.sleep(sleep_retry)
+                continue
+            except requests.exceptions.ConnectionError:
+                msg = f"ConnectionError, waiting {sleep_retry}s before retrying"
+                logger.info(f"[{__parser__}]:execute_query: {msg}", extra={'frontend': str(self.frontend)})
+                time.sleep(sleep_retry)
+                continue
+            if response.status_code != 200:
+                msg = f"Status Code {response.status_code}, waiting {sleep_retry}s before retrying"
+                logger.info(f"[{__parser__}]:execute_query: {msg}", extra={'frontend': str(self.frontend)})
+                time.sleep(sleep_retry)
+                continue
+            break  # no error we break from the loop
+
+        if response is None or response.status_code != 200:
+            if response is None:
+                msg = f"Error Cybereason API Call URL: {url} [TIMEOUT]"
+                logger.error(f"[{__parser__}]:execute_query: {msg}", extra={'frontend': str(self.frontend)})
```

Comment:
shouldn't there be an exception raising here ?
Otherwise, execution will complete without errors